go/teams/sig.go

```diff
@@ -287,20 +287,33 @@ func precheckLinkToPost(ctx context.Context, g *libkb.GlobalContext,
 
 func precheckLinksToPost(ctx context.Context, g *libkb.GlobalContext,
 	sigMultiItems []libkb.SigMultiItem, state *TeamSigChainState, me keybase1.UserVersion) (err error) {
-
 	defer g.CTraceTimed(ctx, "precheckLinksToPost", func() error { return err })()
 
+	// As an optimization, AppendChainLink consumes its state.
+	// We don't consume our state parameter.
+	// So clone state before we pass it along to be consumed.
+	if state != nil {
+		state = state.DeepCopyToPtr()
+	}
+
+	for _, sigItem := range sigMultiItems {
+		state, err = playSigItem(ctx, g, me, state, sigItem)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func playSigItem(ctx context.Context, g *libkb.GlobalContext, me keybase1.UserVersion, state *TeamSigChainState, sigMultiItem libkb.SigMultiItem) (*TeamSigChainState, error) {
 	isAdmin := true
 	if state != nil {
 		role, err := state.GetUserRole(me)
 		if err != nil {
 			role = keybase1.TeamRole_NONE
 		}
 		isAdmin = role.IsAdminOrAbove()
-
-		// As an optimization, AppendChainLink consumes its state.
-		// We don't consume our state parameter.
-		// So clone state before we pass it along to be consumed.
 		state = state.DeepCopyToPtr()
```

Comment:
i don't think we need to deep copy again inside this method. the one in precheckLinksToPost is to protect the caller of precheckLinksToPost from having the state change underneath it. if precheckLinksToPost is going to be the only thing calling playSigItem then i think we can just leave it as a pointer and update it with each pass and actually not even have to return it from this method.